onnx/defs/image/defs.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <functional>
+#include "onnx/defs/data_type_utils.h"
+#include "onnx/defs/schema.h"
+#include "onnx/defs/tensor_proto_util.h"
+
+namespace ONNX_NAMESPACE {
+
+static const char* ImageDecoder_ver20_doc =
+    R"DOC(Loads and decodes and image from a file. If it can't decode for any reason (e.g. corrupted encoded stream, invalid format, it will return an empty matrix).
+
+The following image formats are supported: BMP, JPEG, JPEG2000, TIFF, PNG, WebP, Portable image format (PBM, PGM, PPM, PXM, PNM).
+
+Decoded images follow a channel-last layout: (Height, Width, Channels).
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    ImageDecoder,
+    20,
+    OpSchema()
+        .SetDoc(ImageDecoder_ver20_doc)
+        .Attr(
+            "pixel_format",
+            "Pixel format. Can be one of \"RGB\", \"BGR\", or \"Grayscale\".",
+            AttributeProto::STRING,
+            std::string("RGB"))
+        .Attr(
+            "dtype",
+            "(Optional). Specifies data type for the elements of the output tensor T2. "
+            "Meant for future extensions, currently limited to uint8 only. "
```

Comment:
Why add this now? It seems better that this attribute and the TypeConstraint for T2 be consistent with each other. So, either support other types throughout the spec or omit the attribute?

Reply:
Removed it.